frontend/src/utils/giteaInternalApi.js

```diff
@@ -90,10 +90,19 @@ export const commitInitialFiles = async ({
   newBranchName,
   csrf,
 }) => {
-  // remove any leading "/"
-  let filePaths = files.map(file =>
-    file.path.startsWith('/') ? file.path.substring(1) : file.path,
-  )
+  let filePaths = files.map(file => {
+    let filePath = file.path
+    if (!filePath) {
+      console.warn(
+        'File object in commitInitialFiles does not have a "path" property. Using "name" instead:',
+        file.name,
+      )
+      filePath = file.name
+    }
```

Comment:
This is the main change. We can't guarantee that `file.path` is there and not an empty string. Right now the only place where we saw `file.path === ""` was in Cypress on Electron (hence we saw the failures on CI) but we will need to do some more checks on different browsers. I checked unstable and beta chrome and it was fine, so I'm not sure what's happening with this API -- react-dropzone is also being very clever and filling it in when it can...

Anyway, I don't think it's related to any issue you saw with uploading multiple files.